File: tests/testdata/codegen/logical-name-pp/nodejs/logical-name.ts
import * as khulnasoft from "@khulnasoft/khulnasoft";
import * as random from "@khulnasoft/random";

export = async () => {
    const config = new khulnasoft.Config();
    const configLexicalName = config.require("cC-Charlie_charlie.😃⁉️");
    const resourceLexicalName = new random.RandomPet("aA-Alpha_alpha.🤯⁉️", {prefix: configLexicalName});
    return {
        "bB-Beta_beta.💜⁉": resourceLexicalName.id,
        "dD-Delta_delta.🔥⁉": resourceLexicalName.id,
    };
}

File: tests/testdata/codegen/logical-name-pp/python/logical-name.py
import khulnasoft
import khulnasoft_random as random

config = khulnasoft.Config()
config_lexical_name = config.require("cC-Charlie_charlie.😃⁉️")
resource_lexical_name = random.RandomPet("aA-Alpha_alpha.🤯⁉️", prefix=config_lexical_name)
khulnasoft.export("bB-Beta_beta.💜⁉", resource_lexical_name.id)
khulnasoft.export("dD-Delta_delta.🔥⁉", resource_lexical_name.id)
